pkg/executor/executor.go

```diff
@@ -91,11 +90,12 @@ func DoBuild(k KanikoBuildArgs) (name.Reference, v1.Image, error) {
 			if err != nil {
 				return nil, nil, err
 			}
-			auth, err := authn.DefaultKeychain.Resolve(ref.Context().Registry)
+			k8sc, err := k8schain.NewInCluster(k8schain.Options{})
```

Comment:
`NewInCluster` will use the pod's serviceaccount identity to try and talk to the API server and read secrets, you want [this](https://github.com/google/go-containerregistry/blob/1fcd4c5f05065afeab866058b03c688d0ab89157/pkg/authn/k8schain/k8schain.go#L121-L131) which I now realize isn't documented.  Oops.

Reply:
Oh okay. I'll give that a try. Sounds like we need better docs about which is useful in which scenarios, and maybe a rename, since I figured `NewInCluster` would be the one I wanted while trying to auth from in the cluster.

Reply:
That worked! 🎉